sdk/openfeature-nodejs-provider/src/DevCycleProvider.ts

```diff
@@ -0,0 +1,277 @@
+     * Resolve a string OpenFeature flag and its evaluation details.
+     * @param flagKey
+     * @param defaultValue
+     * @param context
+     */
+    async resolveStringEvaluation(
+        flagKey: string,
+        defaultValue: string,
+        context: EvaluationContext
+    ): Promise<ResolutionDetails<string>> {
+        return this.getDVCVariable(flagKey, defaultValue, context)
+    }
+
+    /**
+     * Resolve a number OpenFeature flag and its evaluation details.
+     * @param flagKey
+     * @param defaultValue
+     * @param context
+     */
+    async resolveNumberEvaluation(
+        flagKey: string,
+        defaultValue: number,
+        context: EvaluationContext
+    ): Promise<ResolutionDetails<number>> {
+        return this.getDVCVariable(flagKey, defaultValue, context)
+    }
+
+    /**
+     * Resolve a object OpenFeature flag and its evaluation details.
+     * @param flagKey
+     * @param defaultValue
+     * @param context
+     */
+    async resolveObjectEvaluation<T extends JsonValue>(
+        flagKey: string,
+        defaultValue: T,
+        context: EvaluationContext
+    ): Promise<ResolutionDetails<T>> {
+        return this.getDVCVariable(flagKey, this.defaultValueFromJsonValue(defaultValue), context)
+    }
+
+    /**
+     * Convert a OpenFeature JsonValue default value into DVCJSON default value for evaluation.
+     * @param jsonValue
+     * @private
+     */
+    private defaultValueFromJsonValue(jsonValue: JsonValue): DVCJSON {
+        if (typeof jsonValue !== 'object' || Array.isArray(jsonValue)) {
+            throw new Error('DevCycle only supports object values for JSON flags')
+        }
+        if (!jsonValue) {
+            throw new Error('DevCycle does not support null default values for JSON flags')
+        }
+
+        // Hard casting here because our DVCJSON typing enforces a flat object when we actually support
+        // a JSON Object of any depth. Will be fixed soon.
+        return jsonValue as DVCJSON
+    }
+
+    /**
+     * Convert a DVCVariable result into a OpenFeature ResolutionDetails.
+     * TODO: add support for variant / reason / and more error codes from DVC.
+     * @param variable
+     * @private
+     */
+    private resultFromDVCVariable<T>(variable: DVCVariable): ResolutionDetails<T> {
+        return {
+            value: variable.value as T,
+            reason: variable.isDefaulted
+                ? StandardResolutionReasons.DEFAULT
+                : StandardResolutionReasons.TARGETING_MATCH,
+        }
+    }
+
+    /**
+     * Convert an OpenFeature EvaluationContext into a DVCUser.
+     * @param context
+     * @private
+     */
+    private dvcUserFromContext(context: EvaluationContext): DVCUser {
+        const user_id = context.targetingKey ?? context.user_id
+        if (!user_id) throw new Error('Missing targetingKey or user_id in context')
+        if (typeof user_id !== 'string') throw new Error('targetingKey or user_id must be a string')
```

Comment:
Similar to [here](https://github.com/DevCycleHQ/js-sdks/pull/483/files#r1182788383), you may find it helpful to throw a [InvalidContextError](https://github.com/open-feature/js-sdk/blob/main/packages/shared/src/errors/invalid-context-error.ts) or [TargetingKeyMissingError](https://github.com/open-feature/js-sdk/blob/main/packages/shared/src/errors/targeting-key-missing-error.ts).

See https://openfeature.dev/specification/types#error-code for the semantic means of each.